distributed/diagnostics/tests/test_worker_plugin.py

```diff
@@ -284,3 +284,19 @@ def teardown(self, worker):
 
     await c.submit(inc, 0)
     assert w.bar == 456
+
+
+@gen_cluster(nthreads=[], client=True)
+async def test_race(c, s):
+    class MyPlugin(WorkerPlugin):
+        async def setup(self, worker):
+            await asyncio.sleep(0.2)  # create some space for tasks to start
```

Comment:
Not sure how we feel about this timing-based test. It won't fail in CI, but if a regression occurs, it may then pass with false positives and become flaky instead of clearly failing.

It would probably be feasible to make this non-timing-based using some Events.

Reply:
It would probably be feasible to do that.  I think that it would make the test significantly more complicated and hard to understand.  I also don't plan to spend the time to do that (this isn't high on my priority list).  Instead, I'd rather increase the timing limit.  I think that 200ms is _more_ than enough to reliably trigger a failure 95% of the time.  I don't think that this is going to slip under our radar.